src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryObjectInfo.cs

```diff
@@ -618,7 +618,8 @@ internal Type[] GetMemberTypes(string[] inMemberNames, Type objectType)
             bool isFound = false;
             for (int i = 0; i < _cache._memberInfos.Length; i++)
             {
-                if (!memberMissing && inMemberNames[i].Equals(_cache._memberInfos[i].Name))
+                if (!memberMissing && 
+                    inMemberNames[i].Equals(FormatterServices.GetFullMemberInfoName(_cache._memberInfos[i], objectType)))
```

Comment:
Looking at the reference source, the code being changed is the same as it is in desktop.  Why does corefx need these changes?  I'm wondering if the root problem is actually somewhere else and this change is just a Band-Aid masking the actual issue.